packages/agent/src/canisterStatus/index.ts

```diff
@@ -87,8 +87,8 @@ export const request = async (options: {
         const response = await agent.readState(canisterId, {
           paths: [encodedPaths[index]],
         });
-        const cert = new Certificate(response, agent);
-        const verified = await cert.verify();
+        const cert = new Certificate(response.certificate, agent.fetchRootKey());
```

Comment:
`agent.fetchRootKey()` is asynchronous, so this would need an `await` - otherwise, the value that gets passed to the Certificate constructor is a Promise.

```typescript
const rootKey = await agent.fetchRootKey();
const cert = new Certificate(response.certificate, rootKey);
```

Reply:
The `Certiificate` constructor takes a `Promise`, it's later awaited on in `Certificate.verify`, where it's used - I didn't want to block before the point where it was needed (not that it makes much of a difference in practice, as the user pretty much has to call `verify` to do anything with the certificate).